tests/compile-fail/borrow_box.rs

```diff
@@ -0,0 +1,14 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(clippy)]
+#![allow(boxed_local)]
+#![allow(blacklisted_name)]
+
+pub fn test(foo: &Box<bool>) { //~ ERROR you seem to be trying to use `&Box<T>`
```

Comment:
After thinking about it, `let foo: &Box<T>` should probably not be linted about:

```rust
let foo: &Box<T> = some_extern_crate::bar();
```
It's not your fault if the crate returns that. You can't change it.